File: src/dataset/python/food-chain_exercism-ThomasZumsteg/claude/ClaudeSonnet4_food_chain_v1.py
def recite(start_verse, end_verse):
    return [verse(v) for v in range(start_verse-1, end_verse)]


animals = (
    ("fly", "", ""),
    ("spider", "It wriggled and jiggled and tickled inside her.",
        " that wriggled and jiggled and tickled inside her"),
    ("bird", "How absurd to swallow a bird!", ""),
    ("cat", "Imagine that, to swallow a cat!", ""),
    ("dog", "What a hog, to swallow a dog!", ""),
    ("goat", "Just opened her throat and swallowed a goat!", ""),
    ("cow", "I don't know how she swallowed a cow!", ""),
    ("horse", "She's dead, of course!", ""),)


end = "I don't know why she swallowed the fly. Perhaps she'll die."

# Pre-compute verse fragments to avoid repeated string operations
_verse_cache = {}

def _build_verse_fragments():
    fragments = {}
    for v in range(8):
        animal_name, reaction, suffix = animals[v]
        opening = f"I know an old lady who swallowed a {animal_name}.{reaction}"
        
        if v == 7:
            fragments[v] = opening
        else:
            catch_lines = []
            for i in range(v, 0, -1):
                catch_lines.append(f"She swallowed the {animals[i][0]} to catch the {animals[i-1][0]}{animals[i-1][2]}.")
            fragments[v] = opening + ''.join(catch_lines) + end
    return fragments

_verse_cache = _build_verse_fragments()

def verse(v):
    return _verse_cache[v]

File: src/dataset/cpp/food-chain_exercism-tqa236/claude/ClaudeSonnet4_food_chain_v1.cpp
#include "food_chain.h"

static const VERSE_INFO song_info[] = {
    {},
    {"fly", "I don't know why she swallowed the fly. Perhaps she'll die."},
    {"spider", "It wriggled and jiggled and tickled inside her."},
    {"bird", "How absurd to swallow a bird!"},
    {"cat", "Imagine that, to swallow a cat!"},
    {"dog", "What a hog, to swallow a dog!"},
    {"goat", "Just opened her throat and swallowed a goat!"},
    {"cow", "I don't know how she swallowed a cow!"},
    {"horse", "She's dead, of course!"},
};

static const char* const SPIDER_SUFFIX = " that wriggled and jiggled and tickled inside her.\n";
static const char* const REGULAR_SUFFIX = ".\n";

std::string food_chain::verse(int num) {
    std::string res;
    res.reserve(512);
    
    res += "I know an old lady who swallowed a ";
    res += song_info[num].animal;
    res += ".\n";

    if (num > FLY) {
        res += song_info[num].reaction;
        res += '\n';
    }

    if (num == HORSE) return res;

    while (num > FLY) {
        res += "She swallowed the ";
        res += song_info[num].animal;
        res += " to catch the ";
        res += song_info[num - 1].animal;
        
        if (--num == SPIDER) {
            res += SPIDER_SUFFIX;
        } else {
            res += REGULAR_SUFFIX;
        }
    }

    res += song_info[FLY].reaction;
    res += '\n';
    
    return res;
}

std::string food_chain::verses(int start, int end) {
    std::string res;
    res.reserve(4096);
    
    for (int i = start; i <= end; i++) {
        res += verse(i);
        res += '\n';
    }
    return res;
}

std::string food_chain::sing() { 
    return verses(1, 8); 
}